### PR TITLE
[fix](mtmv) Fix select literal result wrongly in group by when use materialized view

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes
This is brought by https://github.com/apache/doris/pull/34274

if mv def is 
```sql
select o_orderdate from  orders group by o_orderdate;
```

query is as followiing, the result is wrong. this pr fix the problem.
```sql
select 1 from orders group by o_orderdate;
```

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java [hunk 1]
```diff
@@ -266,6 +266,34 @@ protected LogicalAggregate<Plan> doRewriteQueryByView(
...
+    /**
+     * handle the scene that query top plan not use the group by in query bottom aggregate
+     * If mv is select o_orderdate from  orders group by o_orderdate;
+     * query is select 1 from orders group by o_orderdate.
+     * Or mv is select o_orderdate from  orders group by o_orderdate, o_order_key;
+     * query is select o_orderdate from orders group by o_orderdate;
+     * if the slot which query top project use can not cover the slot which query bottom aggregate group by slot
+     * should compensate group by to make sure the data is right.
+     */
+    private static boolean needCompensateGroupBy(Set<? extends Expression> queryTopPlanGroupBySet,
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java:278]: does queryTopPlanGroupBySet mean top project slots? if it does, we may use a better name such as topProjectList?

**Review by morrySnow — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

**Review by zfr9527 — APPROVED** [review]
LGTM

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run beut

**Comment by morrySnow** [comment]
run cloudut

**Pull request merged**